tfe/resource_tfe_notification_configuration_test.go

```diff
@@ -0,0 +1,436 @@
+		}
+
+		if notificationConfiguration.DestinationType != tfe.NotificationDestinationTypeGeneric {
+			return fmt.Errorf("Bad destination type: %s", notificationConfiguration.DestinationType)
+		}
+
+		if notificationConfiguration.Enabled != true {
+			return fmt.Errorf("Bad enabled value: %t", notificationConfiguration.Enabled)
+		}
+
+		// Token is write only, can't read it
+
+		if !reflect.DeepEqual(notificationConfiguration.Triggers, []string{tfe.NotificationTriggerCreated, tfe.NotificationTriggerNeedsAttention}) {
+			return fmt.Errorf("Bad triggers: %v", notificationConfiguration.Triggers)
+		}
+
+		if notificationConfiguration.URL != "http://example.com/?update=true" {
+			return fmt.Errorf("Bad URL: %s", notificationConfiguration.URL)
+		}
+
+		return nil
+	}
+}
+
+func testAccCheckTFENotificationConfigurationAttributesDuplicateTriggers(notificationConfiguration *tfe.NotificationConfiguration) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		if notificationConfiguration.Name != "notification_duplicate_triggers" {
+			return fmt.Errorf("Bad name: %s", notificationConfiguration.Name)
+		}
+
+		if notificationConfiguration.DestinationType != tfe.NotificationDestinationTypeGeneric {
+			return fmt.Errorf("Bad destination type: %s", notificationConfiguration.DestinationType)
+		}
+
+		if notificationConfiguration.Enabled != false {
+			return fmt.Errorf("Bad enabled value: %t", notificationConfiguration.Enabled)
+		}
+
+		// Token is write only, can't read it
+
+		if !reflect.DeepEqual(notificationConfiguration.Triggers, []string{tfe.NotificationTriggerCreated}) {
+			return fmt.Errorf("Bad triggers: %v", notificationConfiguration.Triggers)
+		}
+
+		if notificationConfiguration.URL != "http://example.com" {
+			return fmt.Errorf("Bad URL: %s", notificationConfiguration.URL)
+		}
+
+		return nil
+	}
+}
+
+func testAccCheckTFENotificationConfigurationDestroy(s *terraform.State) error {
+	tfeClient := testAccProvider.Meta().(*tfe.Client)
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "tfe_notification_configuration" {
+			continue
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No instance ID is set")
+		}
+
+		_, err := tfeClient.NotificationConfigurations.Read(ctx, rs.Primary.ID)
+		if err == nil {
+			return fmt.Errorf("Notification configuration %s still exists", rs.Primary.ID)
+		}
+	}
+
+	return nil
+}
+
+const testAccTFENotificationConfiguration_basic = `
+resource "tfe_organization" "foobar" {
+  name  = "terraform-test"
+  email = "admin@company.com"
+}
+
+resource "tfe_workspace" "foobar" {
+  name         = "workspace-test"
+  organization = "${tfe_organization.foobar.id}"
+}
+
+resource "tfe_notification_configuration" "foobar" {
+  name             			= "notification_basic"
+  destination_type 			= "generic"
+  url              			= "http://example.com"
+  workspace_external_id     = "${tfe_workspace.foobar.external_id}"
```

Comment:
Might want to filter your test fixtures through `terraform fmt`.

Reply:
Done! :)